ci/prow/make_config.go

```diff
@@ -218,320 +219,57 @@ var (
-  ignore_pending: false          # Show in-progress tests
-  column_header:
-  - configuration_value: Commit  # Shows the commit number on column header
-  - configuration_value: infra-commit
-  num_columns_recent: 10         # The number of columns to consider "recent" for a variety of purposes
-  use_kubernetes_client: true    # ** This field is deprecated and should always be true **
-  is_external: true              # ** This field is deprecated and should always be true **
-  alert_stale_results_hours: 26  # Alert if tests haven't run for a day (1 day + 2h)
-  num_failures_to_alert: 1       # Alert for every failure
-  num_passes_to_disable_alert: 1 # Consider a failing test passing if it has 1 or more consecutive passes
-
-default_dashboard_tab:
-  open_test_template:            # The URL template to visit after clicking on a cell
-    url: https://gubernator.knative.dev/build/<gcs_prefix>/<changelist>
-  file_bug_template:             # The URL template to visit when filing a bug
-    url: https://github.com/knative/serving/issues/new
-    options:
-    - key: title
-      value: "Test \"<test-name>\" failed"
-    - key: body
-      value: <test-url>
-  attach_bug_template:           # The URL template to visit when attaching a bug
-    url:                         # Empty
-    options:                     # Empty
-  # Text to show in the about menu as a link to another view of the results
-  results_text: See these results in Gubernator
-  results_url_template:          # The URL template to visit after clicking
-    url: https://gubernator.knative.dev/builds/<gcs_prefix>
-  # URL for regression search links.
-  code_search_path: github.com/knative/serving/search
-  num_columns_recent: 10
-  code_search_url_template:      # The URL template to visit when searching for changelists
-    url: https://github.com/knative/serving/compare/<start-custom-0>...<end-custom-0>
-  alert_options:
-    alert_mail_to_addresses: "knative-productivity-dev@googlegroups.com"
-	`
+	generalTestgridConfig = "testgrid_general_config.yaml"
 
 	// testGroupTemplate is the template for the test group config
-	testGroupTemplate = `
-- name: [[.Base.TestGroupName]]
-  gcs_prefix: [[.GcsLogDir]]
-  [[indent_map 2 .Extras]]
-	`
+	testGroupTemplate = "testgrid_testgroup.yaml"
 
 	// dashboardTabTemplate is the template for the dashboard tab config
-	dashboardTabTemplate = `
-  - name: [[.Name]]
-    test_group_name: [[.Base.TestGroupName]]
-    base_options: "[[.BaseOptions]]"
-    [[indent_map 2 .Extras]]
-	`
+	dashboardTabTemplate = "testgrid_dashboardtab.yaml"
 
 	// dashboardGroupTemplate is the template for the dashboard tab config
-	dashboardGroupTemplate = `
-- name: [[.Name]]
-  dashboard_names:
-  [[indent_array 2 .RepoNames]]
-	`
+	dashboardGroupTemplate = "testgrid_dashboardgroup.yaml"
 )
 
 // Yaml parsing helpers.
 
+// read template yaml file content
+func readTemplate(fp string) string {
+	content, err := ioutil.ReadFile(path.Join(templateDir, fp))
```

Comment:
I can't help saying that this is I/O inefficient. :) You're reading it from disk every time you need a template. You could store them in a map once they're read, so you only read them from disk once. But I'm OK with a TODO if you don't want to do it now, this is not a critical performance issue.

Reply:
done